File: python/triangle_count.py
class Solution:
    """
    @param S: A list of integers
    @return: An integer
    """
    def triangleCount(self, S):
        # write your code here
        S.sort()
        res = 0
        for i in range(2,len(S)):
            k = 0
            j = i-1
            while k < j:
                if S[k] + S[j] > S[i]:
                    res += j-k
                    j -= 1
                else :
                    k +=1
        
        return res


test = Solution()
S =[3,4,6,7]
res = test.triangleCount(S)
print(res)

File: systemDesign/triangleCount.cpp
class Solution {
public:
    /**
     * @param S: A list of integers
     * @return: An integer
     */
    int triangleCount(vector<int> &S) {
        // write your code here
        sort(S.begin(), S.end());
        int res = 0;
        int n = S.size();
        for(int i = n-1; i>=2; i--) {
            int left =0;
            for(int right = i-1; right > left; right--){
                while(left < right && S[left] + S[right] <= S[i]){
                    left++;
                }
                res += right-left;
               
            }
        }
        return res;

    }
};
